pandas/io/formats/format.py

```diff
@@ -1310,7 +1310,9 @@ def _format(x):
                     tpl = " {v}"
                 fmt_values.append(tpl.format(v=_format(v)))
 
-        fmt_values = _trim_zeros_float(str_floats=fmt_values, decimal=".")
+        fmt_values = _trim_zeros_float(
```

Comment:
there is a larger issue to address first. these are not floats at all. these are object dtypes that are reprering. *why* are they hitting this path is the issue. we likley *do* not want to touch the formatters.

Reply:
there are likley getting *inferred* as floats and that is why the formatter is picked. this is incorrect.

Reply:
This is the `GenericArrayFormatter` - seems reasonable for object type data. I think hitting `_trim_zeros_float` in `GenericArrayFormatter` makes sense because of a case like 
```
s = pd.Series([1.20, "1.00"])
repr(s)
````
which should hit `_trim_zeros_float` so that 1.20 is truncated to 1.2.

Reply:
ok, sure but a *string* should not be truncated. that's where the issue is.

Reply:
i see the issue here.

instead of _trim_zeros_float being a function that accepts a ndarray/list of strings, just have it process a *single* value, then call it when needed in the loop on L1299.

this makes the code much simpler. i am not entirely sure why it was done this way.

Reply:
I think the reason why it was done that way is that floats are formatted with fixed-width, e.g.
```
s = pd.Series([1.230, 1.20000])
repr(s)
```
gives 
```
0    1.23
1    1.20
dtype: float64
```
`_trim_zeros_float` was designed for usage in `FloatArrayFormatter` (so it needs the whole list to keep fixed-width, since it only truncates if all values can be truncated). The usage of this function in `GenericArrayFormatter` is what caused the regression. I think then it makes sense to keep `_trim_zeros_float` as is for that original purpose, and `GenericArrayFormatter` can use a much simpler function (say `_trim_zeros_single_float`) for truncation since it does not have the same fixed width requirements.

Reply:
ok that sounds fine, tests are pretty sensitive here so you can refactor.

Reply:
Thanks for the help here, this is much cleaner!